### PR TITLE
cleanup: bigquery_rest without grpc_utils for cmake

### LINKED ISSUE #12210: Stop building protos for bigquery rest component.
**EDIT:**

We actually need to:

- [x] Stop using `DebugString(Status, TracingOptions)` in `experimental-bigquery_rest`. And remove the dep on `grpc_utils` in this PR.
- [ ] modify the `GOOGLE_CLOUD_CPP_ENABLE_GRPC` logic to also respect `experimental-bigquery_rest`

### PR DESCRIPTION
**Pull request by dbolduc**
Fixes #12210

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12220)

### CODE REVIEW

File: cmake/GoogleCloudCppFeatures.cmake [hunk 1]
```diff
@@ -223,10 +225,13 @@ function (google_cloud_cpp_enable_cleanup)
 
     list(REMOVE_DUPLICATES GOOGLE_CLOUD_CPP_ENABLE)
 
-    set(GOOGLE_CLOUD_CPP_ENABLE_GRPC ON)
-    if ("${GOOGLE_CLOUD_CPP_ENABLE}" STREQUAL "storage")
-        set(GOOGLE_CLOUD_CPP_ENABLE_GRPC OFF)
-    endif ()
+    set(GOOGLE_CLOUD_CPP_ENABLE_GRPC OFF)
+    foreach (feature ${GOOGLE_CLOUD_CPP_ENABLE})
```
Comment (coryan) [inline cmake/GoogleCloudCppFeatures.cmake:229]: Consider: `foreach (feature IN LISTS GOOGLE_CLOUD_CPP_ENABLE)`

File: cmake/GoogleCloudCppFeatures.cmake [hunk 2]
```diff
@@ -23,6 +23,8 @@ include(CreateBazelConfig)
 set(GOOGLE_CLOUD_CPP_LEGACY_FEATURES
     "bigtable;bigquery;iam;logging;pubsub;spanner;storage")
 
+set(GOOGLE_CLOUD_CPP_REST_ONLY_FEATURES "storage;experimental-bigquery_rest")
```
Comment (coryan) [inline cmake/GoogleCloudCppFeatures.cmake:26]: Maybe leave a comment explaining why `compute` and `sqladmin` cannot be in that list?

File: cmake/GoogleCloudCppFeatures.cmake (hunk 1 above)
Comment (coryan) [inline cmake/GoogleCloudCppFeatures.cmake:229]: It seems we could do something like:

```cmake
set(grpc_features ${GOOGLE_CLOUD_CPP_ENABLE})
list(REMOVE_ITEM grpc_features ${GOOGLE_CLOUD_CPP_REST_ONLY_FEATURES})
set(GOOGLE_CLOUD_CPP_ENABLE_GRPC OFF)
if (grpc_features)
  set(GOOGLE_CLOUD_CPP_ENABLE_GRPC ON)
endif ()
```

Needs some testing though.

**Review by coryan — APPROVED** [review]

File: cmake/GoogleCloudCppFeatures.cmake (hunk 1 above)
Comment (dbolduc) [reply cmake/GoogleCloudCppFeatures.cmake]: Fixed the source I copied from.

File: cmake/GoogleCloudCppFeatures.cmake (hunk 2 above)
Comment (dbolduc) [reply cmake/GoogleCloudCppFeatures.cmake:30]: Done.

File: cmake/GoogleCloudCppFeatures.cmake (hunk 1 above)
Comment (dbolduc) [reply cmake/GoogleCloudCppFeatures.cmake]: Apparently not tested well.

File: cmake/GoogleCloudCppFeatures.cmake [hunk 3]
```diff
@@ -253,7 +261,7 @@ endfunction ()
 # Most of them are subdirectories in `google/cloud/`. Some number of them have
 # additional samples that are enabled if needed.
 function (google_cloud_cpp_enable_features)
-    foreach (feature ${GOOGLE_CLOUD_CPP_ENABLE})
+    foreach (feature IN LISTS ${GOOGLE_CLOUD_CPP_ENABLE})
```
Comment (coryan) [inline cmake/GoogleCloudCppFeatures.cmake:264]: If you use `IN LISTS` you want the name of the lists, not their values, i.e. you want `GOOGLE_CLOUD_CPP_ENABLE`, not  `${GOOGLE_CLOUD_CPP_ENABLE}`

Comment (dbolduc) [reply cmake/GoogleCloudCppFeatures.cmake:264]: :monkey: 

D'oh. Thanks.

**Review by coryan — APPROVED** [review]

**Pull request merged**